src/device/pf_cmrpc_helpers.c

```diff
@@ -0,0 +1,55 @@
+#include "pf_includes.h"
+/*********************************************************************
+ *        _       _         _
+ *  _ __ | |_  _ | |  __ _ | |__   ___
+ * | '__|| __|(_)| | / _` || '_ \ / __|
+ * | |   | |_  _ | || (_| || |_) |\__ \
+ * |_|    \__|(_)|_| \__,_||_.__/ |___/
+ *
+ * www.rt-labs.com
+ * Copyright 2018 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+/**
+ * @file
+ * @brief Helper functions for CMRPC
+ *
+ * Located in a separate file to enabling mocking during tests.
+ *
+ */
+
+
+/**
+ * @internal
+ * Generate an UUID, inspired by version 1, variant 1
+ *
+ * See RFC 4122
+ *
+ * @param timestamp        In:   Some kind of timestamp
+ * @param session_number   In:   Session number
+ * @param mac_address      In:   MAC address
+ * @param p_uuid           Out:  The generated UUID
+ */
+void pf_generate_uuid(
+   uint32_t                timestamp,
```

Comment:
Why not fetch the timestamp in this function?

Reply:
In order to have a pure function (no side effects and identical output for same input), and to make it easy to unit test it without mocking.